java/flight/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -103,6 +107,11 @@ public void awaitTermination() throws InterruptedException {
   /** Request that the server shut down. */
   public void shutdown() {
     server.shutdown();
+    try {
+      AutoCloseables.close(resources);
```

Comment:
If server.shutdown() fails, we will leak resources (fail to invoke AutoCloseables.close)?

Reply:
We could, but server.shutdown doesn't throw any checked exceptions. I can move it into a finally block though.

Reply:
Er, no, I can't, since I don't want to shut down the executor before the server itself shuts down.